tests/bsim/bluetooth/mesh/src/test_cdp1.c

```diff
@@ -0,0 +1,430 @@
+	.nvnd = 0,
+};
+
+static const struct bt_mesh_comp_p1_elem test_p1_elem2 = {
+	.nsig = 1,
+	.nvnd = 0,
+};
+
+static const struct bt_mesh_comp_p1_elem test_p1_elem3 = {
+	.nsig = 1,
+	.nvnd = 0,
+};
+
+static const struct bt_mesh_comp_p1_elem test_p1_elem4 = {
+	.nsig = 1,
+	.nvnd = 0,
+};
+
+static struct bt_mesh_comp_p1_elem test_p1_elems[] = {
+	test_p1_elem0, test_p1_elem1, test_p1_elem2, test_p1_elem3, test_p1_elem4,
+};
+
+static void provision_and_configure(struct bt_mesh_test_cfg cfg)
+{
+	int err;
+	uint8_t status;
+
+	err = bt_mesh_provision(net_key, 0, 0, 0, cfg.addr, cfg.dev_key);
+	if (err) {
+		FAIL("Provisioning failed (err %d)", err);
+		return;
+	}
+
+	err = bt_mesh_cfg_cli_app_key_add(0, cfg.addr, 0, 0, app_key, &status);
+	if (err || status) {
+		FAIL("AppKey add failed (err %d, status %u)", err, status);
+		return;
+	}
+}
+
+static void verify_model_item(struct bt_mesh_comp_p1_model_item *mod_item, int elem_idx,
+			      int mod_idx, int offset)
+{
+	ASSERT_EQUAL(test_p1_elem_models[elem_idx][mod_idx + offset].cor_present,
+		     mod_item->cor_present);
+	ASSERT_EQUAL(test_p1_elem_models[elem_idx][mod_idx + offset].format, mod_item->format);
+	ASSERT_EQUAL(test_p1_elem_models[elem_idx][mod_idx + offset].ext_item_cnt,
+		     mod_item->ext_item_cnt);
+	if (mod_item->cor_present) {
+		ASSERT_EQUAL(test_p1_elem_models[elem_idx][mod_idx + offset].cor_id,
+			     mod_item->cor_id);
+	}
+}
+
+static void verify_ext_item(struct bt_mesh_comp_p1_ext_item *ext_item, int elem_idx, int mod_idx,
+			    int offset)
+{
+	struct bt_mesh_comp_p1_ext_item *test_p1_ext_item;
+
+	if (elem_idx == 0 && (mod_idx + offset) == 3) {
+		test_p1_ext_item = &test_p1_ext_mod2_mod1;
+	} else if (elem_idx == 0 && (mod_idx + offset) == 4) {
+		test_p1_ext_item = &test_p1_ext_mod3_mod4;
+	} else if (elem_idx == 0 && (mod_idx + offset) == 5) {
+		test_p1_ext_item = &test_p1_ext_vnd1_mod1;
+	} else if (elem_idx == 4 && (mod_idx + offset) == 0) {
+		test_p1_ext_item = &test_p1_ext_mod6_mod1;
+	} else {
+		return;
+	}
+
+	ASSERT_EQUAL(test_p1_ext_item->type, ext_item->type);
+	if (ext_item->type == SHORT) {
+		ASSERT_EQUAL(test_p1_ext_item->short_item.elem_offset,
+			     ext_item->short_item.elem_offset);
+		ASSERT_EQUAL(test_p1_ext_item->short_item.mod_item_idx,
+			     ext_item->short_item.mod_item_idx);
+	} else {
+		ASSERT_EQUAL(test_p1_ext_item->long_item.elem_offset,
+			     ext_item->long_item.elem_offset);
+		ASSERT_EQUAL(test_p1_ext_item->long_item.mod_item_idx,
+			     ext_item->long_item.mod_item_idx);
+	}
+}
+
+static void verify_cdp1(struct bt_mesh_comp_p1_elem *p1_elem,
+			struct bt_mesh_comp_p1_model_item *mod_item,
+			struct bt_mesh_comp_p1_ext_item *ext_item,
+			struct net_buf_simple *p1_dev_comp)
+{
+	int elem_idx = 0;
+
+	while (bt_mesh_comp_p1_elem_pull(p1_dev_comp, p1_elem)) {
+		ASSERT_EQUAL(test_p1_elems[elem_idx].nsig, p1_elem->nsig);
+		ASSERT_EQUAL(test_p1_elems[elem_idx].nvnd, p1_elem->nvnd);
+
+		int mod_idx, ext_mod_idx;
```

Comment:
I would declare the loop variables in the for-loop(s) instead to keep their reach limited to just within the loop. Just to be safe.
`for (int mod_idx = 0; mod_idx < p1_elem->nsig; mod_idx++) {`

Reply:
Fixed!